r/R/compute.R

```diff
@@ -110,15 +114,16 @@ max.ArrowDatum <- function(..., na.rm = FALSE) {
   scalar_aggregate("min_max", ..., na.rm = na.rm)$GetFieldByName("max")
 }
 
-scalar_aggregate <- function(FUN, ..., na.rm = FALSE) {
+scalar_aggregate <- function(FUN, ..., na.rm = FALSE, na.min_count = 0) {
   a <- collect_arrays_from_dots(list(...))
-  if (!na.rm && a$null_count > 0 && (FUN %in% c("mean", "sum"))) {
-    # Arrow sum/mean function always drops NAs so handle that here
-    # https://issues.apache.org/jira/browse/ARROW-9054
-    return(Scalar$create(NA_real_))
+  if (FUN %in% c("mean", "sum") && !na.rm) {
+    na.min_count = length(a)
+  }
+  if (FUN == "min_max" && na.rm && a$null_count == length(a)) {
+    Array$create(data.frame(min=Inf, max=-Inf))
```

Comment:
Why this special case? Is there a reason the C++ library shouldn't handle this? (Leave a comment explaining and/or linking to the JIRA issue to do this in C++)

Reply:
I'm not sure this is the best place to put this wart :).